### PR TITLE
add mkdir

### PR DESCRIPTION
**Pull request by heapwolf**

### CODE REVIEW

File: fs/index.js [hunk 1]
```diff
@@ -251,6 +251,28 @@ export function lstat (path, options, callback) {
...
+
+  if (typeof mode !== 'number') {
+    throw new TypeError('mode must be a number.')
+  }
+
+  if (mode < 0 || !Number.isFinite(mode)) {
+    throw new RangeError('mode must be a positive finite number.')
+  }
+
+  ipc.request('fsMkdir', { mode, path, recursive }).then((result) => {
```
Comment (jwerle) [inline fs/index.js]: we should handle a rejection with `.catch((err) => callback(err))`

**Review by jwerle — CHANGES_REQUESTED** [review]

**Review by jwerle — APPROVED** [review]
we can add an implementation for `fs.promises.mkdir()` soon too

File: fs/index.js [hunk 2]
```diff
@@ -251,6 +251,29 @@ export function lstat (path, options, callback) {
...
+  }
+
+  const mode = options.mode || 0o777
+  const recursive = options.recurisve === true
+
+  if (typeof mode !== 'number') {
+    throw new TypeError('mode must be a number.')
+  }
+
+  if (mode < 0 || !Number.isFinite(mode)) {
```
Comment (chicoxyzzy) [inline fs/index.js]: `Number.isInteger(mode)` could fit better

**Review by chicoxyzzy — APPROVED** [review]

File: test/src/fs/index.js [hunk 3]
```diff
@@ -116,7 +116,20 @@ test('fs.lchown', async (t) => {})
...
 test('fs.lstat', async (t) => {})
-test('fs.mkdir', async (t) => {})
+test('fs.mkdir', async (t) => {
+  const filename = Math.random().toString(16).slice(2)
+
+  await new Promise((resolveMkdir, reject) => {
+    fs.mkdir(filename, {}, (err) => {
+      if (err) reject(err)
+
+      fs.stat(dirname, (err) => {
```
Comment (chicoxyzzy) [inline test/src/fs/index.js:166]: test is failing because `dirname` is not defined

**Pull request merged**